src/goto-harness/recursive_initialization.cpp

```diff
@@ -1008,6 +1009,27 @@ code_blockt recursive_initializationt::build_function_pointer_constructor(
   for(const auto &target : targets)
   {
     auto const assign = code_assignt{dereference_exprt{result}, target};
+    auto const sym_to_lookup =
+      target.id() == ID_address_of
+        ?
+        // This is either address of or pointer; in pointer case, we don't
+        // need to do anything. In the address of case, the operand is
+        // a symbol representing a target function.
+        to_symbol_expr(to_address_of_expr(target).object()).get_identifier()
+        : "";
+    // skip referencing globals because the corresponding symbols in the symbol
+    // table are no longer marked as file local.
+    if(has_prefix(id2string(sym_to_lookup), FILE_LOCAL_PREFIX))
```

Comment:
The thinking here is that symbols prefixed this way might actually be referenceable, but we decided to push off potential different handling for this case off as a different feature. This is coming from `--export-file-local-symbols` apparently.